cmake/CMakeLists.txt

```diff
@@ -75,6 +75,12 @@ set(REPO_ROOT ${PROJECT_SOURCE_DIR}/..)
 set(ONNXRUNTIME_ROOT ${PROJECT_SOURCE_DIR}/../onnxruntime)
 file (STRINGS "${REPO_ROOT}/VERSION_NUMBER" VERSION_NUMBER)
 
+if (onnxruntime_GCC32)
+  set (CMAKE_SYSTEM_PROCESSOR "x86")
+  set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -m32 -Wno-narrowing")
+  set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -m32 -Wno-narrowing")
+endif()
+
```

Comment:
Can we put CMAKE_SYSTEM_PROCESSOR=x86 and "-m32" into build.py? So that we don't need to add a new cmake option 'onnxruntime_GCC32'

Reply:
GCC implies Linux only -- if we decide to keep the flag, would   **onnxruntime_32bit** be clearer? (i.e. not tied to GCC compiler only)

Reply:
Also, we need more options by default for the x86 build. Should have -msse2 so that Eigen can generate better code. And need -fpmath=sse so that scalar float and double code use SSE instead of slow x87 paths.

Reply:
Adding the flags in next commit.